RUFAS/input_manager.py

```diff
@@ -2221,25 +2221,212 @@ def _validate_properties(self) -> None:
+                    info_map,
+                )
+                raise ValueError
+            if minimum_length is not None and len(default) < minimum_length:
+                om.add_error(
+                    "Invalid metadata default array length.",
+                    f"Invalid 'default' for '{key_path}': 'default' length of {default} is "
+                    f"less than 'minimum_length' length {minimum_length}",
+                    info_map,
+                )
+                raise ValueError
+            if maximum_length is not None and len(default) > maximum_length:
+                om.add_error(
+                    "Invalid metadata default array length.",
+                    f"Invalid 'default' for '{key_path}': 'default' length of {default} is "
+                    f"greater than 'maximum' length {maximum_length}",
+                    info_map,
+                )
+                raise ValueError
+        if maximum_length is not None and minimum_length is not None and maximum_length < minimum_length:
+            om.add_error(
+                "Invalid metadata array length range.",
+                f"Invalid length 'range' for key '{key_path}': 'minimum_length' value {minimum_length} is "
+                f"greater than 'maximum_length' value {maximum_length}",
+                info_map,
+            )
+            raise ValueError
+
+    def _metadata_object_validator(self, key_path: list[str], value: dict[str, Any]) -> None:
+        """Validates object type properties in metadata."""
+        required_object_property_keys = {"type"}
+        valid_object_property_keys = {"type", "description"}
+        self._validate_metadata_properties_keys(
+            required_object_property_keys, valid_object_property_keys, value, key_path
+        )
 
-    def _metadata_object_validator(self, key_path: List[str], value: dict[str, Any]) -> None:
-        """Validator function for object type properties in metadata."""
-        pass
+    def _validate_metadata_properties_keys(
+        self,
+        required_properties_keys: set[str],
+        valid_properties_keys: set[str],
+        properties: dict[str, Any],
+        path: list[str],
+    ) -> None:
+        """Validates that keys in the metadata properties sections."""
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self._validate_metadata_properties_keys.__name__,
+        }
+        if missing_required_keys := required_properties_keys - properties.keys():
+            om.add_error(
+                "Metadata Validation",
+                f"Missing required keys {sorted(missing_required_keys)} for {path}. Required"
+                f" keys are {sorted(required_properties_keys)}.",
+                info_map,
+            )
+            raise ValueError
+        property_type = properties.get("type", "Unknown type")
```

Comment:
```suggestion
        property_type = properties.get("type")
```
If the type is missing or invalid, won't it get caught in `_validate_properties` and not make it to this line?

Reply:
It would right now with the way things are set up but I thought adding this check would make this function less dependent on that initial check in `_validate_properties()` in case there is later on a different way this function gets called.